### DISCUSSION TITLE
--memory option in queue worker command seems to be ignored

### DISCUSSION THREAD

**Discussion by ghost** [laravel/framework#37155] [Q&A]
Im still facing the same or similar issue. I used the --memory parameter together with supervisor and the work command like this:
command=php /var/www/artisan queue:work --sleep=1 --timeout=100 --memory=512

However, a quite big job still fails with the "Allowed memory size of 134217728 bytes exhausted (tried to allocate 5988352 bytes)". What makes me thing that the memory option does not work is the message. It says "of 134217728 byes" which is 128MB, which is the default value of the memory parameter. In the php.ini memory_limit is also set to 512M.

_Originally posted by @DeroSavage in https://github.com/laravel/framework/issues/25738#issuecomment-828387238_

**Comment by paras-malhotra** [comment]
@DeroSavage, the issue is with your php.ini memory limit. Make sure you set this limit in the CLI php.ini file (not FPM).

**Reply by ghost** [reply]
...facepalm x100. Thanks that was it.

**Reply by SWAT10101** [reply]
> @DeroSavage, the issue is with your php.ini memory limit. Make sure you set this limit in the CLI php.ini file (not FPM).

I facing the same issue [here](https://github.com/Maatwebsite/Laravel-Excel/discussions/3430#discussion-3673709)
I change the value in `php.ini` but it does not solve the issue still getting `Allowed memory size`

**Reply by jobs-git** [reply]
it says use the cli path not the fpm path

**Comment by carcinocron** [comment]
For those of you arriving from the future, the change I made was to add `-d memory_limit=512M` like:

```sh
command=php -d memory_limit=512M /var/www/artisan queue:work --sleep=1 --timeout=100 --memory=512
```

I was trying to solve the problem that `"Allowed memory size of [...] bytes exhausted (tried to allocate 5988352 bytes)".` seems to fail silently and nothing gets logged, and in manually triggering `Allowed memory size of [...] bytes exhausted` synthetically I'm able to completely fill the servers RAM way beyound 512M and lock up the server until the OOM kicks in or AWS autobalancer kills the EC2.

After adding `-d memory_limit=512M`, I receive a useful error message in Sentry and in my normal linux log files, the worker dies at the expected memory ceiling.

**Comment by driade** [comment]
For those of you arriving from the future too.  As per config https://github.com/laravel/framework/blob/11.x/src/Illuminate/Queue/Console/WorkCommand.php the default memory limit is 128M, despite of what you've configured in your php.ini.

It's 10 years working with Laravel and I didn't know. This should be stated somehow in the docs for sure; the fact that it dies "silently" also doesn't help too much. I'll try to open a PR in the official docs.

**Reply by driade** [reply]
Someone did it in the past https://github.com/laravel/docs/pull/9719

**Reply by driade** [reply]
Nevertheless, looking carefully at the code seems that the the worker is killed after processing de job. Then, as far as I understand, php.ini options would be used but if the script consumes more than X megabytes (128 by default), the process would be killed. But the job would have been successfully processed, which is great.
